autopush/router/simple.py

```diff
@@ -188,21 +194,42 @@ def _send_notification(self, uaid, node_id, notification):
                               "version": notification.version,
                               "data": notification.data})
         url = node_id + "/push/" + uaid
-        d = self.ap_settings.agent.request(
+        request = self.ap_settings.agent.request(
             "PUT",
             url.encode("utf8"),
             bodyProducer=FileBodyProducer(StringIO(payload)),
         )
-        d.addCallback(IgnoreBody.ignore)
-        return d
+        # Only time out actual calls, not mocks.
+        if isinstance(request, defer.Deferred):
+            timeout = reactor.callLater(self.ap_settings.router_timeout,
+                                        request.cancel)
```

Comment:
I guess I had assumed there'd be away to set a socket-level timeout, rather than having to cancel it from Python. Am I reading this correctly that twisted's HTTP agent has no way to set a time-out on the socket connection portion?